#Introductions

Name: Ansari Indrawan

I'm a data analyst at The Look, one of the largest e-commerce platforms on Mars. I am responsible for the creation of evaluation reports on the sales activities of The Look.

#Objective
To evaluate the sales activities and transactions of The Look e-commerce by applying statistical concepts and business understanding.

#Problem Statement

1. What is the number of transactions with completed status each month until quarter 3 in 2022? What is the average revenue per quarter?
2. What are the total sales per month until the 3rd quarter of 2022? If there is an increase, what are the average sales per quarter?
3. How many users will make transactions with a Completed status each month from quarter 1 through quarter 3 of 2022?
4. What is the most purchased product category each month through the 3rd quarter of 2022?
5. What is the product category that generates the most revenue each month through the 3rd quarter of 2022?
6. What is the relationship between the number of users making transactions by comparing the number of transactions to the total revenue? Select the most related

#Import libraries

In [ ]:
from google.cloud import bigquery
from google.colab import auth

# Connect BigQuery with Colab

In [ ]:
auth.authenticate_user()
print('Authenticated')

project_id = "lat1-hacktive8"
client = bigquery.Client(project=project_id)

Authenticated


# Answering problem 1
The Look E-commerce transactions in 2022

In [ ]:
#menyimpan hasil data query pada variabel p1

p1 = client.query('''


#memilih kolom kolom dari tabel orders

SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dari data kolom order_id
#menampilkan jumlah transaksi pada kolom transaction
#manmpilkan jumlahh item dalam setiap transaksi pada kolom total item sold

    COUNT(DISTINCT order_id) AS transaction,


#menunjukan sumber data dari tabel ordes

FROM `bigquery-public-data.thelook_ecommerce.orders`


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilka dataset dari variabel p1
p1

year  month  quarterly  transaction
0  2022      1          1          742
1  2022      2          1          720
2  2022      3          1          881
3  2022      4          2          856
4  2022      5          2          949
5  2022      6          2          924
6  2022      7          3         1032
7  2022      8          3         1140
8  2022      9          3         1147

##insight problem 1
Based on the table above, we can see data from the number of transactions each month for 3 quarters in 2022. There was an increase in the number of transactions from the first quarter to the third quarter. The lowest transaction occurred in the first quarter with a total of 720 transactions, and the highest transaction occurred in the third quarter which amounted to 1147 transactions. The average number of transactions per quarter is 781 in quarter 1, 909 in quarter 2, and 1106 in quarter 3.

# Answering Problem 2
The Look E-commerce sales in 2022

In [ ]:
#menyimpan hasil data query pada variabel p2

p2 = client.query('''


#memilih kolom kolom dari tabel

SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#CONCAT digunakan untuk menggabungkan nilai total_sales dengan string'USD'
#ROUND digunakan untuk membulatkan hail pada kolom total_sales menjadi hanya 2 angka dibelakang koma
#SUM digunakan untuk menghitung total penjualan dari kolom sale price

    CONCAT(ROUND(SUM(sale_price), 2 ), ' ','USD') AS total_sales,


#memilih kolom dari tabel oerder items

FROM `bigquery-public-data.thelook_ecommerce.order_items`


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p2
p2

year  month  quarterly    total_sales
0  2022      1          1   67617.65 USD
1  2022      2          1   62862.36 USD
2  2022      3          1   74691.14 USD
3  2022      4          2   71157.74 USD
4  2022      5          2    80874.8 USD
5  2022      6          2    86017.1 USD
6  2022      7          3   89794.31 USD
7  2022      8          3    98879.5 USD
8  2022      9          3  101128.68 USD

## insight problem 2
Based on the table above, we can see the data of total sales each month until the third quarter of 2022. There was an increase in total sales from the first quarter to the third quarter, with the lowest sales occurring in the first quarter of the second month, which amounted to 67617.65 USD, and the highest transaction occurred in the third quarter of the ninth month, which amounted to 101128.68 USD. The average of The look's total sales per quarter was found to be 68,390.38 USD in quarter 1, 79,349.88 USD in quarter 2, 96,600.83 USD in quarter 3.

# Answer to Problem 3
User analysis of The Look E-commerce in 2022

In [ ]:
#menyimpan hasil data query pada variabel p3

p3 = client.query('''


SELECT

#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data perkuartal dengan kolom quarterl dari data kolom created_at

    EXTRACT(year FROM created_at) AS year,
    EXTRACT(month FROM created_at) AS month,
    EXTRACT(quarter FROM created_at) AS quarterly,


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dalam setiap group dari kolom user_id
#menampilkan jumlah transaksi pada kolom user_transaction

    COUNT(DISTINCT user_id) AS user_transaction,


#memilih kolom dari tabel oerder items

FROM `bigquery-public-data.thelook_ecommerce.order_items`


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

WHERE status = 'Complete'
    AND EXTRACT(year FROM created_at) = 2022
    AND EXTRACT (quarter FROM created_at) BETWEEN 1 AND 3


#mengelompkan hasil query
#mengurutkan baris berdasarkan bulan

GROUP BY year,month, quarterly
ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p3
p3

year  month  quarterly  user_transaction
0  2022      1          1               761
1  2022      2          1               732
2  2022      3          1               900
3  2022      4          2               865
4  2022      5          2               958
5  2022      6          2               936
6  2022      7          3              1044
7  2022      8          3              1153
8  2022      9          3              1158

## insight problem 3
Based on the table above, we can see data from the number of users who successfully transact in The Look e-commerce every month until the third quarter of 2022. Transacting users experienced an increase from the first quarter to the third quarter, with the lowest transaction occurring in the first quarter of the second month, which amounted to 761 users, while the highest occurred in the third quarter of the ninth month, which amounted to 1158 users. The average number of users who transacted each quarter was 797 in quarter 1, 919 in quarter 2, and 1118 users in quarter 3.

# Answering Problem 4
Analyze The Look's E-commerce products in 2022

In [ ]:
#menyimpan hasil data query pada variabel p4

p4 = client.query('''


#memilih kolom kolom dari hasil query 2
#dengan mengubah nama kolom kategory menjadi most_bought

SELECT

    year,
    month,
    quarterly,
    category as most_bought,
    transaction_amount


#menunjukan sumber data dari tabel hasil query 2

FROM (


#query 2
#memilih kolom kolom dari hasil query 1

      SELECT
          category,
          year,
          month,
          quarterly,
          transaction_amount,


#fungsi ini digunakan untuk memberikan ranking pada setiap baris dalam hasil query 1
#dengan diatur pada tahun 2022 dan setiap bulan pada kuartal 1-3 kemudian diurutkan berdasarkan tabel transaction amount dari yang terbesar
#hasil perankingan disimpan dalam kolom ranking

          RANK() OVER (PARTITION BY year, month ORDER BY transaction_amount DESC) AS ranking


#menunjukan sumber data dari tabel hasil query 1

      FROM (


#query 1
#memilih kolom kolom dari tabel hasil gabungan antara kolom product_id dari order items dan kolom id dari product

            SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

                p.category,
                EXTRACT(year FROM created_at) AS year,
                EXTRACT(month FROM created_at) AS month,
                EXTRACT(quarter FROM created_at) AS quarterly,
                COUNT(category) AS transaction_amount


#menunjukan sumber data dari tabel hasil JOIN antara order items dan product
#JOIN berfungsi untuk menggabungkan baris dari kolom spesifik yang memiliki nilai yang sama

            FROM `bigquery-public-data.thelook_ecommerce.order_items` AS oi
            JOIN `bigquery-public-data.thelook_ecommerce.products` AS p
            ON oi.product_id = p.id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

            WHERE status = 'Complete'
                AND EXTRACT(year FROM created_at) = 2022
                AND EXTRACT(quarter FROM created_at) BETWEEN 1 AND 3


#mengelompokan hasil query

            GROUP BY category, year, month, quarterly
          )


#mentapkan kriteria untuk data yang diquery, dimana baris yang ditampilkan hanya yg berperingkat 1
#mengurutkan baris berdasarkan bulan

      )WHERE ranking = 1
       ORDER BY month ASC;


''').to_dataframe()

In [ ]:
##menampilkan dataset dari variabel p4
p4

year  month  quarterly  most_bought  transaction_amount
0  2022      1          1    Intimates                  81
1  2022      2          1  Tops & Tees                  82
2  2022      3          1    Intimates                 119
3  2022      4          2        Jeans                  97
4  2022      5          2    Intimates                 111
5  2022      6          2        Jeans                 108
6  2022      7          3        Jeans                 110
7  2022      8          3    Intimates                 121
8  2022      9          3    Intimates                 138

##insight problem 4
Based on the table above, we can see the data of total products purchased by category at The Look e-commerce every month until the 3rd quarter of 2022. After the ranking of the products, the ranking of the product category is obtained every month, where only the first rank of each month is displayed based on the highest sales level. In the first quarter, the most purchased product is the Intimates category with 119 transactions. In the second quarter, the most purchased product was also the Intimates category with 111 transactions. For the third quarter, the most purchased product is the Intimates category, which is also the category with the most transactions for 3 quarters, 131 transactions.

# Answering Problem 5
Sales analysis of The Look E-commerce in 2022

In [ ]:
#menyimpan hasil data query pada variabel p4

p5 = client.query('''


#memilih kolom kolom dari hasil query 1
#dengan mengubah nama kolom category menjadi most_sold

SELECT
    year,
    month,
    quarterly,
    category as most_sold,
    revenue,
    amount_sales


#menunjukan sumber data dari tabel hasil query 2

FROM (


#query2
#memilih kolom kolom dari hasil query 1

      SELECT
          category,
          year,
          month,
          quarterly,
          number_of_product,
          revenue,
          amount_sales,


#fungsi ini digunakan untuk memberikan ranking pada setiap baris dalam hasil query 1
#dengan diatur pada tahun 2022 dan setiap bulan pada kuartal 1-3 kemudian diurutkan berdasarkan tabel total product dari yang terbesar
#hasil perankingan disimpan dalam kolom ranking

      RANK() OVER (PARTITION BY year, month ORDER BY number_of_product DESC) AS ranking


#menunjukan sumber data dari tabel hasil query 1

      FROM (


#query 1
#memilih kolom kolom dari tabel hasil gabungan antara kolom product_id dari order items dan kolom id dari product

            SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at
#menampilkan jumlah product pada kolom total_product
#ROUND digunakan untuk membulatkan hasil pada kolom amount_sales menjadi hanya 2 angka dibelakang koma
#SUM digunakan untuk menghitung total penjualan dari kolom retail price, cost, dan sale price
#menambahkan kolom revenue dari hasil pengurangan total tabel retail price dan total cost

                p.category,
                EXTRACT(YEAR FROM created_at) AS year,
                EXTRACT(MONTH FROM created_at) AS month,
                EXTRACT(quarter FROM created_at) AS quarterly,
                COUNT(category) AS number_of_product,
                ROUND(SUM(retail_price)-SUM(cost), 2) AS revenue,
                ROUND(SUM(sale_price), 2) AS amount_sales


#menunjukan sumber data dari tabel hasil JOIN antara order items dan product
#JOIN berfungsi untuk menggabungkan tabel order_items pada kolom product id dengan tabel products dengan menggunakan kolom id.

            FROM `bigquery-public-data.thelook_ecommerce.order_items` AS oi
            JOIN `bigquery-public-data.thelook_ecommerce.products` AS p
            ON oi.product_id = p.id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

            WHERE status = 'Complete' AND EXTRACT(YEAR FROM created_at) = 2022
                AND EXTRACT(QUARTER FROM created_at) BETWEEN 1 AND 3


#mengelompokan hasil query

            GROUP BY category, year, month, quarterly
            )


#mentapkan kriteria untuk data yang diquery, dimana baris yang ditampilkan hanya yg berperingkat 1
#mengurutkan baris berdasarkan bulan

      )WHERE ranking = 1
       ORDER BY month ASC;


''').to_dataframe()

In [ ]:
#menampilkan dataset dari variabel p5
p5

year  month  quarterly    most_sold  revenue  amount_sales
0  2022      1          1    Intimates  1240.92       2674.49
1  2022      2          1  Tops & Tees  1722.82       3903.36
2  2022      3          1    Intimates  1771.83       3749.28
3  2022      4          2        Jeans  4618.07       9844.96
4  2022      5          2    Intimates  1719.88       3680.24
5  2022      6          2        Jeans  5120.20      10977.35
6  2022      7          3        Jeans  5111.15      10989.05
7  2022      8          3    Intimates  2000.58       4257.67
8  2022      9          3    Intimates  2438.83       5238.20

##insight problem 5
Based on the table above, we can see the data of the total products that provide a lot of profit based on categories in The Look e-commerce every month until the 3rd quarter of 2022. After the ranking of products, the ranking of product categories is obtained every month, where only the first rank of each month is displayed based on the highest level of profit results. In the first quarter, the most profitable product was the Tops & Tees category, which amounted to 1722.82 USD transactions. In the second quarter, the most purchased product was the Jeans category with an amount of 5120.20 USD. For the third quarter, the most purchased product is the Jeans category with an amount of 5111.15 USD. The Look's total sales for 3 quarters based on the sales amount column was obtained at 49,015.60 USD and the total profit reached 27,144.28 USD.

# Answer Problem 6
The look E-commerce correlation analysis

In [ ]:
p6 = client.query('''

#memilih kolom kolom dari hasil query 1

SELECT

#CORR digunakan untuk menhitung koefisien korelasi pertama antara kolom jumlah user dan jumlah transaksi,
#kedua antara jumlah user dan total revenue

    CORR(number_of_user, transaction_amount) AS user_transaction_correlation,
    CORR(number_of_user, total_revenue) AS user_revenue_correlation


#menunjukan sumber data dari tabel hasil query 1

FROM (
      SELECT


#Fungsi EXTRACT digunakan untuk mengambil nilai dari bagian tertentu dari kolom created_at, yaitu tahun, bulan, dan kuartal.
#menampilkan data tahun dengan kolom year, data bulan dengan kolom month, data tiap kuartal dengan kolom quarterl dari data kolom created_at

          EXTRACT(year FROM o.created_at) AS year,
          EXTRACT(month FROM o.created_at) AS month,
          EXTRACT(quarter FROM o.created_at) AS quarterly,


#fungsi COUNT untuk menghitung semua data, dengan DISTINCT digunakan untuk menghitung jumlah transaksi unik dalam setiap group dari kolom user_id dan order id dari tabel order items
#menampilkan jumlah transaksi pada kolom user_transaction
#SUM digunakan untuk menghitung total pendapatan dari kolom sale price

          COUNT(DISTINCT oi.user_id) AS number_of_user,
          COUNT(DISTINCT oi.order_id) AS transaction_amount,
          SUM(sale_price) AS total_revenue


#menunjukan sumber data dari tabel hasil JOIN antara order items dan orders
#JOIN berfungsi untuk menggabungkan tabel order_items dengan tabel orders dengan menggunakan kolom order_id.

      FROM bigquery-public-data.thelook_ecommerce.order_items AS oi
      JOIN bigquery-public-data.thelook_ecommerce.orders AS o
      ON oi.order_id = o.order_id


#Menetapkan kriteria untuk data yang akan diquery.Baris pertama membatasi data dengan status Complete
#membatasi untuk data pada tahun 2022 saja
#membatasi data pada kuartal 1 sampai kuartal 3

      WHERE o.status = 'Complete'
          AND EXTRACT(year FROM o.created_at) = 2022
          AND EXTRACT(quarter FROM o.created_at) BETWEEN 1 AND 3


#mengelompkan hasil query

        GROUP BY year, month, quarterly
        )


''').to_dataframe()

In [ ]:
p6

user_transaction_correlation  user_revenue_correlation
0                      0.999909                   0.97271

## insight problem 6
Based on the results of the correlation calculation by comparing the correlation between the number of users and the number of transactions and the correlation between the number of users and the total revenue in TheLook e-commerce, it is possible to observe the shopping patterns of the users. In the User Transaction Correlation column, a correlation value of 0.9999 is obtained, which shows an interrelated relationship between the number of users and the number of transactions. It can be interpreted that as the number of users increases, the number of transactions also tends to increase. Similarly, as shown in the User Revenue Correlation column, the correlation value is 0.9727, which means that as the number of users increases, the total revenue will also tend to increase.

By choosing to focus on the correlation results between the number of users and the number of transactions, which have a high correlation, this shows that users of TheLook e-commerce will make more transactions when the number of users increases. Thus, TheLook's goal is to increase the number of users so that it can significantly increase TheLook e-commerce transactions.

# My problem conclusion

1. The problem set I defined can all be achieved and can be measured in terms of the value displayed in the table as well as the insight provided in each table of analysis results presented in answering each problem.

2. Based on the results of the analysis performed, it is very difficult to reach 250,000 USD at the beginning of the 4th quarter based on the number of transactions and the number of users for 3 quarters. Based on the product category sales, the total revenue for the most purchased product is 27,144.28 USD in 3 quarters, and also based on the sales, the average revenue per quarter for 3 quarters is 68,890 USD, 79,849 USD, 96,600 USD. So it would be unreasonable to target 250,000 USD in sales at the beginning of the 4th quarter when the sales in the 3rd quarter alone is only 96,600 USD.

3. Based on the results of the analysis performed with the 6 problem statements made, it is determined that
- The transaction trend of TheLook has been increasing every quarter by looking at the average value of transactions per quarter for 3 quarters, namely 781 transactions, 909 transactions, 1106 transactions.
- The number of users transacting on TheLook has also increased significantly each quarter for 3 quarters by looking at the average value of each quarter with the number of transactions being 797 users, 919 users, 1118 users.
- The gross merchandise value in 3 quarters is obtained based on the sales results for 3 quarters, which are averaged for each quarter, namely 68,890 USD, 79,849 USD, 96,600 USD. From this value, it can be seen that the trend of sales for 3 quarters has been increasing.                                                
Based on the three criteria explained above, the company TheLook deserves to get new investors because for 3 quarters TheLook's business has experienced a positive trend.